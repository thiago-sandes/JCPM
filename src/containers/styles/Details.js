import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    backgroundColor: '#fff',
    flex: 1,
    paddingTop: '15%',
  },

  loader: {
    flex: 1,
    justifyContent: 'center',
  },

  button: {
    marginVertical: '2%',
    alignSelf: 'center',
  },
});
