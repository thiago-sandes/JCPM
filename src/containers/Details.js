import React, {useEffect, useState} from 'react';
import {View, ActivityIndicator, ScrollView} from 'react-native';

// API
import {getData} from '../api/api';

// Styles
import styles from './styles/Details';

// Components
import MovieInfo from '../components/MovieInfo';
import CustomizableButton from '../components/CustomizableButton';

const Details = ({route, navigation}) => {
  const {imdbID} = route.params;
  const [details, setDetails] = useState(null);

  // Handle API call
  const handleApiCall = async () => {
    const response = await getData({i: imdbID});
    setDetails(response);
  };

  // Handling lifecycle
  useEffect(() => {
    handleApiCall();
    return () => {
      setDetails(null);
    };
  }, []);

  if (details !== null) {
    return (
      <View style={styles.container}>
        <ScrollView>
          <MovieInfo details={details} />
          <CustomizableButton
            styles={styles.button}
            onPress={navigation.goBack}
          />
        </ScrollView>
      </View>
    );
  } else {
    return (
      <View style={styles.loader}>
        <ActivityIndicator size="large" color="#0000ff" />
      </View>
    );
  }
};

export default Details;
