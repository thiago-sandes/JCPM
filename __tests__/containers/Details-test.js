import React from 'react';
import renderer from 'react-test-renderer';
import Details from '../../src/containers/Details';

jest.useFakeTimers();

function fakeTimer(callback) {
  setTimeout(() => {
    callback && callback();
  }, 1000);
}

test('Details snapshot', async () => {
  const mockedProps = {
    route: {params: {imdbID: '1234'}},
    navigation: {goBack: jest.fn()},
  };

  const tree = renderer.create(<Details {...mockedProps} />).toJSON();

  fakeTimer();

  expect(tree).toMatchSnapshot();
});
