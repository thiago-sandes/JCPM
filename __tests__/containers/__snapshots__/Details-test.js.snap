// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#0000ff"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
